### ISSUE TITLE
Figure out how to write about R packages

### ISSUE THREAD

**Issue by arilamstein** [RConsortium/censusguide#7]
In the section of the Guide where we write about R packages, we need to decide how to talk about them.

One obvious way is to copy the CRAN description. But this has a few problems.

1. Many of the authors of packages the Guide covers have written multiple packages. CRAN's "directory" structure doesn't handle suites of packages very well. We should highlight, where appropriate, when multiple packages were intended to be used together.

2. CRAN favors a dry, technical description of its packages. That is useful, but has drawbacks. For example, I suspect that many of the authors of packages we are covering saw a particular feature-set lacking in existing packages, and view their package as plugging that hole. I think that the Guide should capture some of this "color commentary".

I should also note that it's possible that some of these packages have overlapping or even duplicative functionality. While we don't want to harp on that in the Guide, it would be useful to understand that as part of the process of writing the Guide.

**Comment by EAVWing** [comment]
You are so spot on with this.

**Comment by gergness** [comment]
The DESCRIPTION wouldn't be great for ipumsr either because US Census data is only part of what we offer. I can't speak for all package authors, but I'd be happy to write a blurb so that you didn't have to try to do everything yourself.

**Comment by arilamstein** [comment]
> The DESCRIPTION wouldn't be great for ipumsr either because US Census data is only part of what we offer. I can't speak for all package authors, but I'd be happy to write a blurb so that you didn't have to try to do everything yourself.

This would be great @gergness ! Then I could use that as an example to email to other package authors. Thank you!

**Comment by gergness** [comment]
Sure! How would you like it, PR into csv? And any guidance on length?

**Comment by arilamstein** [comment]
Good question. Maybe create a new directory, "package-descriptions", and
create a markdown file like "ipumsr.md". I would say aim for 1-2 paragraphs.

**Comment by arilamstein** [comment]
I just completed a draft of the part of the Guide where I introduce people to the various datasets that Census publishes. You can read it [here](https://github.com/RConsortium/censusguide/blob/master/draft.md). A problem there was that Census runs over 100 censuses, surveys and programs, but I only wanted to single out the "top" ones to discuss, and then link to the rest. In that situation @loganpowell and I decided to look at API request traffic. We found that the top 5 programs accounted for the vast majority of requests. 

This seems relevant to figuring out how to figure out how to talk about R packages. I have a list 22 packages on CRAN that relate to US Census Data. I don't think it benefits readers of the Guide to include all of them. The Guide is intended to be more of a survey / primer than an exhaustive document. And I think that including a list of all 22 packages in alphabetical order (as CRAN does) would not be serving the intended audience well.

As an exploration on other ways to do this, I just wrote a script that uses the `cranlogs` package to look at the number of downloads each of the 22 package has gotten over the last year, and then calculate the percent of downloads each package has. You can view the script, along with its output [here](https://github.com/RConsortium/censusguide/blob/master/package-downloads.R).

For the sake of completeness, I am also including the output of the script here:
````
#    package            annual_downloads percent
#    <chr>                         <dbl>   <dbl>
#  1 tigris                        40646      22
#  2 acs                           39077      21
#  3 choroplethr                   31225      17
#  4 tidycensus                    15044       8
#  5 choroplethrMaps               13383       7
#  6 censusapi                      7260       4
#  7 ipumsr                         5857       3
#  8 UScensus2010                   4280       2
#  9 noncensus                      4206       2
# 10 UScensus2000tract              3278       2
# 11 census                         2918       2
# 12 idbr                           2856       2
# 13 UScensus2000cdp                2800       1
# 14 totalcensus                    2524       1
# 15 censusGeography                2013       1
# 16 SeerMapper                     1529       1
# 17 SeerMapperWest                 1506       1
# 18 SeerMapperRegs                 1481       1
# 19 SeerMapper2010Regs             1477       1
# 20 SeerMapper2010East             1475       1
# 21 SeerMapperEast                 1473       1
# 22 SeerMapper2010West             1468       1
````

So if we want to follow the same "highlight a few and then link to an exhaustive list" approach here, we could use this list to inform that discussion.

Another point to make is that many of these entries should, in the guide, probably be lumped together. For example, `UScensus2000dbp` and `UScensus2000tract` are part of the same effort. 

Another point to make is that I am unclear if the various `Seer` packages should be mentioned in the Guide at all. I have been in contact with the package author about this.

**Comment by arilamstein** [comment]
I'm closing this because I think that the main theoretical work is done. I just updated the draft to include the CRAN descriptions of the top 7 packages. In the case of IPUMS I switched the CRAN description for what @gergness wrote. 

There is a placeholder link in the draft for a list to all 22 packages. 

I should also probably put, in writing in the repository, some kind of explanation / criteria for when to include a package in the main document vs. the appendix. I can imagine this becoming a touchy subject for package authors ("Why is my package in the appendix but package x is in the main article?") and right now I am making it up as I go along. 

Three things that stand out at this moment are that:

1. The difference in downloads between packages 5 and 6 is pretty stark: 13k / 7% to 7k / 4%. Based on this one could imagine an argument to say "only highlight the top 5 packages in the main document".
2. However I consider package 6 to be significant and worthy of inclusion because it is the only package on the list that attempts to provide complete coverage of the Census API, which currently has over 200 endpoints. Most R packages in the top 5 focus on the ACS and Decennial Census, however this is far from all that Census does.
3. Similarly, package 7 provides access to census data through IPUMS, which is a significant resource for demographic research. While I have not worked with IPUMS myself, I have met people in the field who prefer to get US Census data from IPUMS as opposed to the Census website. 

So if I had to create a guideline now I would say "a significant percentage of downloads or addressing a significant hole in the "literature" of CRAN packages..
